### DISCUSSION TITLE
Changing a component instance to another component

### DISCUSSION THREAD

**Discussion by iwoithe** [uxbox/uxbox#965] [Q&A] [answered]
Frequently while creating prototypes, I duplicate artboards. However, a lot of the time I want to change a component instance to another component. It starts getting annoying having to delete the item and adding in a new one and configuring it again. Is there a way to change a component instance to another component in penpot?

Thanks

**Comment by hirunatan** [comment, marked as answer]
Hi, @iwoithe. Changing an instance to any general component would be a very complex operation, with a lot of use cases. We are working on a more concrete solution, that would be some kind of components with variants. So you may have some different states in a same component (e.g. open/closed, pushed/unpushed). We think this would fit most use cases for what you are asking.

But this is already in an early state of thinking.

**Reply by iwoithe** [reply]
> But this is already in an early state of thinking.

Good to know. Thanks :slightly_smiling_face:
